server/accounts.go

```diff
@@ -4370,41 +4509,108 @@ func (tr *transform) getHashPartition(key []byte, numBuckets int) string {
+				transformedToken = tr.getHashPartition(keyForHashing, int(tr.dtokmfintargs[i]))
+			case Wildcard: // simple substitution
+				transformedToken = tokens[tr.dtokmftokindexesargs[i][0]]
+			case SplitFromLeft:
+				sourceToken := tokens[tr.dtokmftokindexesargs[i][0]]
+				sourceTokenLen := len(sourceToken)
+				position := int(tr.dtokmfintargs[i])
+				if position > 0 && position < sourceTokenLen {
+					transformedToken = sourceToken[:position] + "." + sourceToken[position:]
+				} else { // too small to split at the requested position: don't split
+					transformedToken = sourceToken
+				}
+			case SplitFromRight:
+				sourceToken := tokens[tr.dtokmftokindexesargs[i][0]]
+				sourceTokenLen := len(sourceToken)
+				position := int(tr.dtokmfintargs[i])
+				if position > 0 && position < sourceTokenLen {
+					transformedToken = sourceToken[:sourceTokenLen-position] + "." + sourceToken[sourceTokenLen-position:]
+				} else { // too small to split at the requested position: don't split
+					transformedToken = sourceToken
+				}
+			case SliceFromLeft:
+				sourceToken := tokens[tr.dtokmftokindexesargs[i][0]]
+				sourceTokenLen := len(sourceToken)
+				sliceSize := int(tr.dtokmfintargs[i])
+				if sliceSize > 0 && sliceSize < sourceTokenLen {
+					transformedToken = ""
+					for i := 0; i+sliceSize <= sourceTokenLen; i += sliceSize {
+						if i != 0 {
+							transformedToken = transformedToken + "."
+						}
+						transformedToken = transformedToken + sourceToken[i:i+sliceSize]
+						if i+sliceSize != sourceTokenLen && i+sliceSize+sliceSize > sourceTokenLen {
+							transformedToken = transformedToken + "." + sourceToken[i+sliceSize:]
+							break
+						}
+					}
+				} else { // too small to slice at the requested size: don't slice
+					transformedToken = sourceToken
+				}
+			case SliceFromRight:
+				sourceToken := tokens[tr.dtokmftokindexesargs[i][0]]
+				sourceTokenLen := len(sourceToken)
+				sliceSize := int(tr.dtokmfintargs[i])
+				if sliceSize > 0 && sliceSize < sourceTokenLen {
+					transformedToken = ""
+					remainder := sourceTokenLen % sliceSize
+					if remainder > 0 {
+						transformedToken = transformedToken + sourceToken[:remainder] + "."
+					}
+					for i := remainder; i+sliceSize <= sourceTokenLen; i += sliceSize {
+						transformedToken = transformedToken + sourceToken[i:i+sliceSize]
+						if i+sliceSize < sourceTokenLen {
+							transformedToken = transformedToken + "."
+						}
+					}
+				} else { // too small to slice at the requested size: don't slice
+					transformedToken = sourceToken
+				}
+			case Split:
+				sourceToken := tokens[tr.dtokmftokindexesargs[i][0]]
+				splits := strings.Split(sourceToken, tr.dtokmfstringargs[i])
+				transformedToken = ""
+				for j, split := range splits {
+					if split != "" {
+						transformedToken = transformedToken + split
+					}
+					if j < len(splits)-1 && splits[j+1] != "" && j != 0 {
+						transformedToken = transformedToken + "."
```

Comment:
can't we just use `b.WriteString()` instead and remove line 4613

Reply:
basically saying replace all assignments of `transformedToken` above with `b.WriteString`

Reply:
Sure, it's small but still an optimization to not use that intermediary variable